core/domain/classifier_services.py

```diff
@@ -100,8 +101,39 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id)
+    state = exploration.states[state_name]
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+        'TextInput']['classifier_id']
+    if state.classifier_model_id:
+        classifier = get_classifier_by_id(state.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            algorithm_id)
+        sc.from_dict(sc_dict)
```

Comment:
Thanks @seanlip, that's a really good point. But, I'd like to raise one point that I had raised in another comment earlier - LDAStringClassifier is most probably going to be discarded. In this case, it makes sense to change the algorithm registry class so that it returns the class itself and not the instance (to follow the life cycle that you have described). The alternative algorithm (probably Naive Bayes) can follow the structure that you have described but I think changing LDAStringClassifier is a waste of time (since we are moving on to a different approach).